### PR TITLE
Case insensitively check acceptance

### PR DESCRIPTION
**Pull request by phiggins**
## Description

Then env and argument strategies checked the value case insensitively while the provided value strategy didn't. This adds tests to all 3 to make this behavior explicit and changes the provided value strategy to match the other two.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/ruby/lib/license_acceptance/strategy/file.rb [hunk 1]
```diff
@@ -28,11 +28,9 @@ def accepted?(product_relationship)
         logger.debug("Searching for the following licenses: #{missing_licenses.map(&:id)}")
 
         searching.each do |product|
-          found = false
           config.license_locations.each do |loc|
             f = ::File.join(loc, product.filename)
             if ::File.exist?(f)
-              found = true
```
Comment (phiggins) [inline components/ruby/lib/license_acceptance/strategy/file.rb:35]: Unrelated cleanup, nothing to see here.

**Review by tyler-ball — APPROVED** [review]
Thank you!
